PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,25 @@
+Please do not create a Pull Request without creating an issue first.
+Any change needs to be discussed before proceeding.
+Failure to do so may result in the rejection of the pull request.
+
+Please provide enough information so that others can review your pull request:
+
+<!-- You can skip this if you're fixing a typo or adding an app to the Showcase. -->
+
+Explain the **details** for making this change. What existing problem does the pull request solve?
+
+<!-- Example: When "Adding a function to do X", explain why it is necessary to have a way to do X. -->
+
+**Test plan (required)**
+
+Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI.
+
+<!-- Make sure tests pass on both Travis and Codacy CI. -->
+
+**Code formatting**
+
+<!-- See the simple style guide. -->
+
+**Closing issues**
+
+Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).
```

Comment:
I personally like the word 'fixes', because closing an issue always doesn't mean it is fixed, but a PR always fixes an issue. I guess the auto-close works for `fixes` also.